service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/EhrExtractStatusService.java

```diff
@@ -115,13 +118,107 @@ public class EhrExtractStatusService {
     private static final String FILENAME_TYPE_PLACEHOLDER = "FILENAME_PLACEHOLDER_ID=";
     private static final String ACKS_SET = ACK_HISTORY + DOT + ACKS;
     public static final int EHR_EXTRACT_SENT_DAYS_LIMIT = 8;
-    private static final String REASON_ERROR_CODE = "04";
-    public static final String REASON_ERROR_MESSAGE = "The acknowledgement has been received after 8 days";
+    private static final String UNEXPECTED_CONDITION_ERROR_CODE = "99";
+    public static final String UNEXPECTED_CONDITION_ERROR_MESSAGE = "The acknowledgement has been received after 8 days";
 
     private final MongoTemplate mongoTemplate;
     private final EhrExtractStatusRepository ehrExtractStatusRepository;
     private final TimestampService timestampService;
 
+    @Scheduled(cron = "${timeout.cronTime}")
+    public void checkForEhrExtractAckTimeouts() {
+        List<EhrExtractStatus> inProgressEhrExtractTransfers = findInProgressTransfers();
+        var now = Instant.now();
+        var ehrExtractStatusWithExceededUpdateLimit = inProgressEhrExtractTransfers
+            .stream()
+            .filter(ehrExtractStatus -> Objects.isNull(ehrExtractStatus.getEhrReceivedAcknowledgement()))
+            .filter(ehrExtractStatus -> hasLastUpdateExceededEightDays(ehrExtractStatus, now))
+            .toList();
```

Comment:
What was your thinking for converting it to a list here and then back to a stream inside of `updateEhrExtractStatusListWithEhrReceivedAcknowledgementError`, seems unnecessary.

Reply:
It's better to keep `ehrExtractStatusWithExceededUpdateLimit` as a List rather than a Stream in this case. This is because you need to check whether the list is empty, and if you use a Stream, applying findAny() (which is a terminal operation) like below:
`if (!ehrExtractStatusWithExceededUpdateLimit.findAny().isEmpty()) {` ...

will consume the stream and make it unusable for further operations. This can lead to a `java.lang.IllegalStateException` if you try to use the stream afterward. By using a List, you can easily check if it's empty and avoid these issues, ensuring that the method operates without exceptions or side effects.

Reply:
> you need to check whether the list is empty

Citation needed.